commit 618bd2c202b4be4aac47fe520a7264bd041a2764
Author: Paul Kocialkowski <paul.kocialkowski@bootlin.com>
Date:   Fri Feb 15 17:17:08 2019 +0100

    net: phy: xgmiitorgmii: Support generic PHY status read
    
    [ Upstream commit 197f9ab7f08ce4b9ece662f747c3991b2f0fbb57 ]
    
    Some PHY drivers like the generic one do not provide a read_status
    callback on their own but rely on genphy_read_status being called
    directly.
    
    With the current code, this results in a NULL function pointer call.
    Call genphy_read_status instead when there is no specific callback.
    
    Fixes: f411a6160bd4 ("net: phy: Add gmiitorgmii converter support")
    Signed-off-by: Paul Kocialkowski <paul.kocialkowski@bootlin.com>
    Signed-off-by: David S. Miller <davem@davemloft.net>
    Signed-off-by: Sasha Levin <sashal@kernel.org>

diff --git a/drivers/net/phy/xilinx_gmii2rgmii.c b/drivers/net/phy/xilinx_gmii2rgmii.c
index 74a8782313cf..bd6084e315de 100644
--- a/drivers/net/phy/xilinx_gmii2rgmii.c
+++ b/drivers/net/phy/xilinx_gmii2rgmii.c
@@ -44,7 +44,10 @@ static int xgmiitorgmii_read_status(struct phy_device *phydev)
 	u16 val = 0;
 	int err;
 
-	err = priv->phy_drv->read_status(phydev);
+	if (priv->phy_drv->read_status)
+		err = priv->phy_drv->read_status(phydev);
+	else
+		err = genphy_read_status(phydev);
 	if (err < 0)
 		return err;
 
diff -uN a/1.txt b/1.txt
--- a/dummy/rpi_1449_618bd2c202b4be4aac47fe520a7264bd041a2764.txt	1970-01-01 00:00:00.000000000 +0000
+++ b/dummy/rpi_1449_618bd2c202b4be4aac47fe520a7264bd041a2764.txt	2013-12-23 04:07:40.000000000 +0000
@@ -0,0 +1 @@
+dummy file to ensure patch has content.
